DocumentsFromSnapshotMigration/src/test/java/org/opensearch/migrations/bulkload/EndToEndTest.java

```diff
@@ -104,6 +108,18 @@ private void migrationDocumentsWithClusters(
             sourceClusterOperations.createIndex(indexName, body);
             targetClusterOperations.createIndex(indexName, body);
 
+            if (includeCompletionFieldIndex && sourceVersion != UnboundVersionMatchers.isBelowES_5_X) {
```

Comment:
Move this check into a function `sourceSupportsCompletionFields()` and use that during the setup and verification steps, then we don't have to tweak the parameter generation logic